src/System.Slices/src/System/MemoryUtils.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static class MemoryUtils
+    {
+        /// <summary>
+        /// platform independent fast memory comparison
+        /// for x64 it is as fast as memcmp of msvcrt.dll, for x86 it is up to two times faster!!
+        /// </summary>
+        internal static bool MemCmp<[Primitive]T>(Span<T> first, Span<T> second)
+            where T : struct
+        {
+            unsafe
+            {
+                // prevent GC from moving memory
+                var firstPinnedHandle = GCHandle.Alloc(first.Object, GCHandleType.Pinned);
+                var secondPinnedHandle = GCHandle.Alloc(second.Object, GCHandleType.Pinned);
+
+                try
+                {
+                    byte* firstPointer = (byte*)PtrUtils.ComputeAddress(first.Object, first.Offset).ToPointer();
+                    byte* secondPointer = (byte*)PtrUtils.ComputeAddress(second.Object, second.Offset).ToPointer();
+
+                    int step = sizeof(void*) * 5;
+                    int totalBytesCount = first.Length * PtrUtils.SizeOf<T>();
+                    byte* firstPointerLimit = firstPointer + (totalBytesCount - step);
+
+                    if (totalBytesCount > step)
+                    {
+                        while (firstPointer < firstPointerLimit)
+                        {   // IMPORTANT: in order to get HUGE benefits of loop unrolling on x86 we use break instead of return
```

Comment:
Do you mean the usage of break instead of return or the whole method itself ? ;)